core/src/main/java/tc/oc/pgm/shops/Shop.java

```diff
@@ -0,0 +1,87 @@
+package tc.oc.pgm.shops;
+
+import static net.kyori.adventure.key.Key.key;
+import static net.kyori.adventure.sound.Sound.sound;
+import static net.kyori.adventure.text.Component.translatable;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import net.kyori.adventure.sound.Sound;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.PlayerInventory;
+import tc.oc.pgm.api.feature.FeatureDefinition;
+import tc.oc.pgm.api.feature.FeatureInfo;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.shops.menu.Category;
+import tc.oc.pgm.shops.menu.Icon;
+import tc.oc.pgm.shops.menu.Payment;
+
+@FeatureInfo(name = "shops")
+public class Shop implements FeatureDefinition {
+
+  private static Sound PURCHASE_SOUND = sound(key("fire.ignite"), Sound.Source.MASTER, 1f, 1.4f);
+
+  private final String id;
+  private final String name;
+  private final ImmutableList<Category> categories;
```

Comment:
```suggestion
  private final List<Category> categories;
```

Reply:
Categories are really a list, you need them indexed. it doesn't make sense to generify them to collection